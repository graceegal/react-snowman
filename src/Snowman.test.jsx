import { it, expect } from "vitest";
import React from "react";

import { render, fireEvent } from "@testing-library/react";
import Snowman from "./Snowman";
import TEST_IMAGES from "./_testCommon.js";

it("renders without crashing", function () {
    render(
        <Snowman
            images={TEST_IMAGES}
            words={["apple"]}
        />
    );
});

it("matches snapshot at initial render", function () {
    const { container } = render(
        <Snowman
            images={TEST_IMAGES}
            words={["apple"]}
        />
    );

    expect(container).toMatchSnapshot();
});

it("Game lost: last image stays on page, losing message is displayed, buttons are not visible", function () {
    const { container, debug } = render(
        <Snowman
            images={TEST_IMAGES}
            words={["apple"]}
        />
    );

    // make 6 wrong letter guesses
    const rBtn = container.querySelector('button[value="r"]');
    fireEvent.click(rBtn);

    const bBtn = container.querySelector('button[value="b"]');
    fireEvent.click(bBtn);

    const qBtn = container.querySelector('button[value="q"]');
    fireEvent.click(qBtn);

    const xBtn = container.querySelector('button[value="x"]');
    fireEvent.click(xBtn);

    const zBtn = container.querySelector('button[value="z"]');
    fireEvent.click(zBtn);

    const yBtn = container.querySelector('button[value="y"]');
    fireEvent.click(yBtn);

    debug();

    expect(
        container.querySelector('img[alt="6"]')
    ).toBeInTheDocument();

    expect(
        container.querySelector('.letter')
    ).toHaveStyle('visibility: hidden');


    expect(
        container.querySelector('.lose-msg')
    ).toHaveStyle('visibility: visible');


});

it("Game lost: matches snapshot", function () {
    const { container, debug } = render(
        <Snowman
            images={TEST_IMAGES}
            words={["apple"]}
        />
    );

    // make 6 wrong letter guesses
    const rBtn = container.querySelector('button[value="r"]');
    fireEvent.click(rBtn);

    const bBtn = container.querySelector('button[value="b"]');
    fireEvent.click(bBtn);

    const qBtn = container.querySelector('button[value="q"]');
    fireEvent.click(qBtn);

    const xBtn = container.querySelector('button[value="x"]');
    fireEvent.click(xBtn);

    const zBtn = container.querySelector('button[value="z"]');
    fireEvent.click(zBtn);

    const yBtn = container.querySelector('button[value="y"]');
    fireEvent.click(yBtn);


    expect(container).toMatchSnapshot();

});



