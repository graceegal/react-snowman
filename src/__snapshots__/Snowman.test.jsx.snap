// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Game lost: last image stays on page, losing message is displayed, buttons are not visible 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
    />
    <p>
      Number wrong: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="letter"
      style="visibility: hidden;"
    >
      <button
        id="a"
        value="a"
      >
        a
      </button>
      <button
        disabled=""
        id="b"
        value="b"
      >
        b
      </button>
      <button
        id="c"
        value="c"
      >
        c
      </button>
      <button
        id="d"
        value="d"
      >
        d
      </button>
      <button
        id="e"
        value="e"
      >
        e
      </button>
      <button
        id="f"
        value="f"
      >
        f
      </button>
      <button
        id="g"
        value="g"
      >
        g
      </button>
      <button
        id="h"
        value="h"
      >
        h
      </button>
      <button
        id="i"
        value="i"
      >
        i
      </button>
      <button
        id="j"
        value="j"
      >
        j
      </button>
      <button
        id="k"
        value="k"
      >
        k
      </button>
      <button
        id="l"
        value="l"
      >
        l
      </button>
      <button
        id="m"
        value="m"
      >
        m
      </button>
      <button
        id="n"
        value="n"
      >
        n
      </button>
      <button
        id="o"
        value="o"
      >
        o
      </button>
      <button
        id="p"
        value="p"
      >
        p
      </button>
      <button
        disabled=""
        id="q"
        value="q"
      >
        q
      </button>
      <button
        disabled=""
        id="r"
        value="r"
      >
        r
      </button>
      <button
        id="s"
        value="s"
      >
        s
      </button>
      <button
        id="t"
        value="t"
      >
        t
      </button>
      <button
        id="u"
        value="u"
      >
        u
      </button>
      <button
        id="v"
        value="v"
      >
        v
      </button>
      <button
        id="w"
        value="w"
      >
        w
      </button>
      <button
        disabled=""
        id="x"
        value="x"
      >
        x
      </button>
      <button
        disabled=""
        id="y"
        value="y"
      >
        y
      </button>
      <button
        disabled=""
        id="z"
        value="z"
      >
        z
      </button>
    </p>
    <p
      class="lose-msg"
      style="visibility: visible;"
    >
      You Lose! Correct Word: 
      apple
    </p>
  </div>
</div>
`;

exports[`Game lost: matches snapshot 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
    />
    <p>
      Number wrong: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="letter"
      style="visibility: hidden;"
    >
      <button
        id="a"
        value="a"
      >
        a
      </button>
      <button
        disabled=""
        id="b"
        value="b"
      >
        b
      </button>
      <button
        id="c"
        value="c"
      >
        c
      </button>
      <button
        id="d"
        value="d"
      >
        d
      </button>
      <button
        id="e"
        value="e"
      >
        e
      </button>
      <button
        id="f"
        value="f"
      >
        f
      </button>
      <button
        id="g"
        value="g"
      >
        g
      </button>
      <button
        id="h"
        value="h"
      >
        h
      </button>
      <button
        id="i"
        value="i"
      >
        i
      </button>
      <button
        id="j"
        value="j"
      >
        j
      </button>
      <button
        id="k"
        value="k"
      >
        k
      </button>
      <button
        id="l"
        value="l"
      >
        l
      </button>
      <button
        id="m"
        value="m"
      >
        m
      </button>
      <button
        id="n"
        value="n"
      >
        n
      </button>
      <button
        id="o"
        value="o"
      >
        o
      </button>
      <button
        id="p"
        value="p"
      >
        p
      </button>
      <button
        disabled=""
        id="q"
        value="q"
      >
        q
      </button>
      <button
        disabled=""
        id="r"
        value="r"
      >
        r
      </button>
      <button
        id="s"
        value="s"
      >
        s
      </button>
      <button
        id="t"
        value="t"
      >
        t
      </button>
      <button
        id="u"
        value="u"
      >
        u
      </button>
      <button
        id="v"
        value="v"
      >
        v
      </button>
      <button
        id="w"
        value="w"
      >
        w
      </button>
      <button
        disabled=""
        id="x"
        value="x"
      >
        x
      </button>
      <button
        disabled=""
        id="y"
        value="y"
      >
        y
      </button>
      <button
        disabled=""
        id="z"
        value="z"
      >
        z
      </button>
    </p>
    <p
      class="lose-msg"
      style="visibility: visible;"
    >
      You Lose! Correct Word: 
      apple
    </p>
  </div>
</div>
`;

exports[`last image stays on page once game is lost 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
    />
    <p>
      Number wrong: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="letter"
      style="visibility: hidden;"
    >
      <button
        id="a"
        value="a"
      >
        a
      </button>
      <button
        disabled=""
        id="b"
        value="b"
      >
        b
      </button>
      <button
        id="c"
        value="c"
      >
        c
      </button>
      <button
        id="d"
        value="d"
      >
        d
      </button>
      <button
        id="e"
        value="e"
      >
        e
      </button>
      <button
        id="f"
        value="f"
      >
        f
      </button>
      <button
        id="g"
        value="g"
      >
        g
      </button>
      <button
        id="h"
        value="h"
      >
        h
      </button>
      <button
        id="i"
        value="i"
      >
        i
      </button>
      <button
        id="j"
        value="j"
      >
        j
      </button>
      <button
        id="k"
        value="k"
      >
        k
      </button>
      <button
        id="l"
        value="l"
      >
        l
      </button>
      <button
        id="m"
        value="m"
      >
        m
      </button>
      <button
        id="n"
        value="n"
      >
        n
      </button>
      <button
        id="o"
        value="o"
      >
        o
      </button>
      <button
        id="p"
        value="p"
      >
        p
      </button>
      <button
        disabled=""
        id="q"
        value="q"
      >
        q
      </button>
      <button
        disabled=""
        id="r"
        value="r"
      >
        r
      </button>
      <button
        id="s"
        value="s"
      >
        s
      </button>
      <button
        id="t"
        value="t"
      >
        t
      </button>
      <button
        id="u"
        value="u"
      >
        u
      </button>
      <button
        id="v"
        value="v"
      >
        v
      </button>
      <button
        id="w"
        value="w"
      >
        w
      </button>
      <button
        disabled=""
        id="x"
        value="x"
      >
        x
      </button>
      <button
        disabled=""
        id="y"
        value="y"
      >
        y
      </button>
      <button
        disabled=""
        id="z"
        value="z"
      >
        z
      </button>
    </p>
    <p
      class="lose-msg"
      style="visibility: visible;"
    >
      You Lose! Correct Word: 
      apple
    </p>
  </div>
</div>
`;

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="0"
      src="[object Object]"
    />
    <p>
      Number wrong: 
      0
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="letter"
      style="visibility: visible;"
    >
      <button
        id="a"
        value="a"
      >
        a
      </button>
      <button
        id="b"
        value="b"
      >
        b
      </button>
      <button
        id="c"
        value="c"
      >
        c
      </button>
      <button
        id="d"
        value="d"
      >
        d
      </button>
      <button
        id="e"
        value="e"
      >
        e
      </button>
      <button
        id="f"
        value="f"
      >
        f
      </button>
      <button
        id="g"
        value="g"
      >
        g
      </button>
      <button
        id="h"
        value="h"
      >
        h
      </button>
      <button
        id="i"
        value="i"
      >
        i
      </button>
      <button
        id="j"
        value="j"
      >
        j
      </button>
      <button
        id="k"
        value="k"
      >
        k
      </button>
      <button
        id="l"
        value="l"
      >
        l
      </button>
      <button
        id="m"
        value="m"
      >
        m
      </button>
      <button
        id="n"
        value="n"
      >
        n
      </button>
      <button
        id="o"
        value="o"
      >
        o
      </button>
      <button
        id="p"
        value="p"
      >
        p
      </button>
      <button
        id="q"
        value="q"
      >
        q
      </button>
      <button
        id="r"
        value="r"
      >
        r
      </button>
      <button
        id="s"
        value="s"
      >
        s
      </button>
      <button
        id="t"
        value="t"
      >
        t
      </button>
      <button
        id="u"
        value="u"
      >
        u
      </button>
      <button
        id="v"
        value="v"
      >
        v
      </button>
      <button
        id="w"
        value="w"
      >
        w
      </button>
      <button
        id="x"
        value="x"
      >
        x
      </button>
      <button
        id="y"
        value="y"
      >
        y
      </button>
      <button
        id="z"
        value="z"
      >
        z
      </button>
    </p>
    <p
      class="lose-msg"
      style="visibility: hidden;"
    >
      You Lose! Correct Word: 
      apple
    </p>
  </div>
</div>
`;

exports[`matches snapshot at initial render 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="0"
      src="[object Object]"
    />
    <p>
      Number wrong: 
      0
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="letter"
      style="visibility: visible;"
    >
      <button
        id="a"
        value="a"
      >
        a
      </button>
      <button
        id="b"
        value="b"
      >
        b
      </button>
      <button
        id="c"
        value="c"
      >
        c
      </button>
      <button
        id="d"
        value="d"
      >
        d
      </button>
      <button
        id="e"
        value="e"
      >
        e
      </button>
      <button
        id="f"
        value="f"
      >
        f
      </button>
      <button
        id="g"
        value="g"
      >
        g
      </button>
      <button
        id="h"
        value="h"
      >
        h
      </button>
      <button
        id="i"
        value="i"
      >
        i
      </button>
      <button
        id="j"
        value="j"
      >
        j
      </button>
      <button
        id="k"
        value="k"
      >
        k
      </button>
      <button
        id="l"
        value="l"
      >
        l
      </button>
      <button
        id="m"
        value="m"
      >
        m
      </button>
      <button
        id="n"
        value="n"
      >
        n
      </button>
      <button
        id="o"
        value="o"
      >
        o
      </button>
      <button
        id="p"
        value="p"
      >
        p
      </button>
      <button
        id="q"
        value="q"
      >
        q
      </button>
      <button
        id="r"
        value="r"
      >
        r
      </button>
      <button
        id="s"
        value="s"
      >
        s
      </button>
      <button
        id="t"
        value="t"
      >
        t
      </button>
      <button
        id="u"
        value="u"
      >
        u
      </button>
      <button
        id="v"
        value="v"
      >
        v
      </button>
      <button
        id="w"
        value="w"
      >
        w
      </button>
      <button
        id="x"
        value="x"
      >
        x
      </button>
      <button
        id="y"
        value="y"
      >
        y
      </button>
      <button
        id="z"
        value="z"
      >
        z
      </button>
    </p>
    <p
      class="lose-msg"
      style="visibility: hidden;"
    >
      You Lose! Correct Word: 
      apple
    </p>
  </div>
</div>
`;
